### PR TITLE
Query Dependency on Entity Checking Added

### PR DESCRIPTION
**Pull request by mehmetfurkanaydin**
Related [https://github.com/sirensolutions/kibi-internal/issues/690](url)

### CODE REVIEW

**Comment by scampi** [comment]
run this command to fix the syntax

```sh
$ grunt eslint:source
```

File: kibiutils.js [hunk 1]
```diff
@@ -11,6 +11,58 @@
     // PRIVATE methods
     //
 
+    var _checkSingleQuery = function (query) {
+
+      var regex = /@doc\[.+?\]@/;
```
Comment (scampi) [inline kibiutils.js]: rename to `entityRegex`

File: kibiutils.js [hunk 2]
```diff
@@ -11,6 +11,58 @@
...
+      var multilineCommentRegex = /\/\*(.|[\r\n])*?\*\//g;
+      var singleLineRegex = /(-- |# |\/\/).*/g;
+
+      // check for sparql and sql queries
+      if ((query.activationQuery && regex.test(query.activationQuery.replace(multilineCommentRegex, '').replace(singleLineRegex, ''))) ||
```
Comment (scampi) [inline kibiutils.js]: add the following check:

```js
if (isSQL() && isSPARQL()) {
  ...
}
```

File: tests.js [hunk 3]
```diff
@@ -170,5 +170,72 @@ describe('slugifyId()', function () {
...
+       'from test \n' +
+        '/* where name \n' + '= \'@doc[_source][github_id]@\' */'
+    ,
+      'select * \n' +
+       'from test \n' +
+       '# where name = \'@doc[_source][github_id]@\''
+    ,
+      'select * \n' +
+       'from test \n' +
+        '/* where name \n' + '= \'@doc[_source] \n ' + '[github_id]@\ */'
```
Comment (scampi) [inline tests.js]: add a query like:

```sql
select *
from test
where id = 123 # @doc[_source][github_id]@
```

File: tests.js [hunk 4]
```diff
@@ -170,5 +170,72 @@ describe('slugifyId()', function () {
...
+  var DependentQueries = [
+     'select * \n' +
+       'from test \n' +
+       ' where name = \'@doc[_source][github_id]@\''
+    ,
+      'select * \n' +
+       'from test \n' +
+        '- where name = \'@doc[_source][github_id]@\''
+    ,
+      'select * \n' +
```
Comment (scampi) [inline tests.js]: why is this query dependent ? The block comment doesn't finish.

The query seems incomplete. I think it should be something like:

```sql
select *
from test
/* where name
= 123 */ where name = '@doc[_source][github_id]@'
```

File: tests.js [hunk 5]
```diff
@@ -170,5 +170,72 @@ describe('slugifyId()', function () {
   it('should do nothing if the id is undefined', function () {
     expect(kibiutils.slugifyId(undefined)).to.be(undefined);
   });
+});
+
+describe('doesQueryDependOnEntity()', function () {
```
Comment (scampi) [inline tests.js]: Add tests for SPARQL queries:

- dependent

```sparql
SELECT *
FROM dbpedia:
WHERE {
  ?s a "@doc[_source][github_id]@"
}
```


- not dependent

```sparql
SELECT *
FROM dbpedia:
WHERE {
  ?s a "Person" # "@doc[_source][github_id]@"
}
```

File: tests.js [hunk 6]
```diff
@@ -170,5 +170,72 @@ describe('slugifyId()', function () {
...
+  ];
+
+  var DependentQueries = [
+     'select * \n' +
+       'from test \n' +
+       ' where name = \'@doc[_source][github_id]@\''
+    ,
+      'select * \n' +
+       'from test \n' +
+        '- where name = \'@doc[_source][github_id]@\''
```
Comment (scampi) [inline tests.js]: What is the leading dash `-` for ?

**Comment by scampi** [comment]
👍

File: tests.js (hunk 4 above)
Comment (mehmetfurkanaydin) [inline tests.js]: I left incomplete for make sure regex is running properly, make sure regex looking beginning end finishing character for comment lines. Ok, I'm changing it now

File: tests.js (hunk 6 above)
Comment (mehmetfurkanaydin) [inline tests.js]: "I left incomplete for make sure regex is running properly, make sure regex looking beginning end finishing character for comment lines. Ok, I'm changing it now"
Same purpose , I'm changing it now too..

**Pull request closed without merging**